src/realm/parser/query_flex.ll

```diff
@@ -17,6 +17,7 @@ unicode "\\u"{hex}{4}
 escape  "\\"[\"\'/bfnrt0\\]
 char1    [^\"\\]
 char2    [^\'\\]
+utf8    [\xC2-\xDF][\x80-\xBF]|(\xE0[\xA0-\xBF]|[\xE1-\xEF][\x80-\xBF])[\x80-\xBF]
```

Comment:
Can you help me understand this? I thought it might be more complex, is this a subset or superset of utf8? (I think we probably don't need to validate people's utf8 so a superset is fine, but if it is only a subset, we might have to document which character sets are not supported)

Reply:
Only having a subset supported would probably be a blocker. I assume just encoding the table found here would be sufficient at this level? https://en.wikipedia.org/wiki/UTF-8 i.e. 1-4 bytes with the boundaries specified and not actually validating whether it is valid utf8 or not.

Reply:
We match up to 3 byte characters which covers "virtually all characters in common use" as the wiki says.

Reply:
Now PR is updated so that we also match 4 byte characters. We match a superset of valid input.

Reply:
Thanks for the clarity, that seems sufficient.